thewalrus/_hafnian.py

```diff
@@ -181,8 +181,42 @@ def find_kept_edges(j, reps):  # pragma: no cover
 
 
 @numba.jit(nopython=True, cache=True)
-def f(A, n):  # pragma: no cover
-    """Evaluate the polynomial coefficients of the function in the eigenvalue-trace formula.
+def powertrace_eigenvalues(E, n):
+    """Evaluate the trace of a matrix with eigenvalues E to the power k, from k=0 to k=n-1.
+
+    Args:
+        E (array): one-dimensional array of eigenvalues
+        n (int): the highest power of the matrix will be n-1
+
+    Returns:
+        array: the trace of the matrix to each power
+    """
+    E_k = E.copy()
+    powtrace = [len(E), E.sum()]
+    for i in range(2, n // 2 + 1):
+        E_k *= E
+        powtrace += [E_k.sum()]
+    return np.array(powtrace, dtype=E.dtype)
```

Comment:
we don't want to use Eigenvalues to calculate powertraces. We should use Labudde.

Reply:
This is only called by `_dist_prob_gray`